### PR TITLE
Fixed bug causing "MAC verified OK" message

### PR DESCRIPTION
**Pull request by kevinclark19a**
## Description

A bug in older versions of openssl was causing the message "MAC verified OK" to be sent out on stderr during the processing of every goal state in the agent. Before PR #2050, the agent would suppress the stderr output. This PR adds that suppression back.

---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and Travis.CI is passing.

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: azurelinuxagent/common/utils/cryptutil.py [hunk 1]
```diff
@@ -88,7 +88,8 @@ def decrypt_p7m(self, p7m_file, trans_prv_file, trans_cert_file, pem_file):
 
             first_proc = subprocess.Popen(first_cmd, stdout=subprocess.PIPE)
 
-            second_proc = subprocess.Popen(second_cmd, stdin=first_proc.stdout, stdout=subprocess.PIPE)
+            second_proc = subprocess.Popen(second_cmd, stdin=first_proc.stdout,
+                stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```
Comment (larohra) [inline azurelinuxagent/common/utils/cryptutil.py:92]: A weird question. The default value is `stderr=None`. If it was None, how were we getting the weird output message in the console? Any idea?

**Review by larohra — APPROVED** [review]
1 small question else LGTM

File: azurelinuxagent/common/utils/cryptutil.py (hunk 1 above)
Comment (kevinclark19a) [reply azurelinuxagent/common/utils/cryptutil.py:92]: My understanding is that `None` defaults to passing through the output to the agent process.

Comment (larohra) [reply azurelinuxagent/common/utils/cryptutil.py:92]: Ahh you're right - 

>stdout=None means that subprocess prints to whatever place your script prints. stdout=PIPE means that subprocess' stdout is redirected to a pipe that you should read e.g., using process.communicate() to read all at once or using process.stdout object to read via a file/iterator interfaces

**Review by pgombar — APPROVED** [review]

**Pull request merged**